iosApp/iosApp/Pages/StopDetails/TripDetailsView.swift

```diff
@@ -140,156 +118,96 @@ struct TripDetailsView: View {
         .onReceive(inspection.notice) { inspection.visit(self, $0) }
         .withScenePhaseHandlers(
             onActive: {
-                if let tripPredictions,
-                   tripPredictionsRepository
-                   .shouldForgetPredictions(predictionCount: tripPredictions.predictionQuantity()) {
-                    self.tripPredictions = nil
-                }
+                stopDetailsVM.returnFromBackground()
                 joinRealtime()
             },
             onInactive: leaveRealtime,
-            onBackground: { leaveRealtime() }
+            onBackground: leaveRealtime
         )
     }
 
     var didLoadData: ((Self) -> Void)?
 
-    @ViewBuilder private func loadingBody() -> some View {
-        TripDetailsStopListView(
-            stops: LoadingPlaceholders.shared.tripDetailsStops(),
-            now: now,
-            onTapLink: { _, _, _ in },
-            routeType: nil
-        )
-        .loadingPlaceholder()
-    }
-
-    private func loadEverything(tripId: String) {
-        loadTripSchedules(tripId: tripId)
-        loadTrip(tripId: tripId)
-    }
-
-    private func loadTripSchedules(tripId: String) {
-        Task {
-            await fetchApi(
-                errorBannerVM.errorRepository,
-                errorKey: "TripDetailsPage.loadTripSchedules",
-                getData: { try await tripRepository.getTripSchedules(tripId: tripId) },
-                onSuccess: { tripSchedulesResponse = $0 },
-                onRefreshAfterError: { loadEverything(tripId: tripId) }
-            )
-        }
-    }
-
-    private func loadTrip(tripId: String) {
-        Task {
-            let response: ApiResult<TripResponse> = try await tripRepository.getTrip(tripId: tripId)
-            let errorKey = "TripDetailsPage.loadTrip"
-            switch onEnum(of: response) {
-            case let .ok(okResponse):
-                errorBannerVM.errorRepository.clearDataError(key: errorKey)
-                trip = okResponse.data.trip
-            case .error:
-                errorBannerVM.errorRepository.setDataError(key: errorKey, action: { loadEverything(tripId: tripId) })
-                trip = nil
+    @ViewBuilder private func tripDetails(
+        _ stops: TripDetailsStopList,
+        _ vehicle: Vehicle?,
+        _ vehicleStop: Stop?,
+        _ route: Route?
+    ) -> some View {
+        let vehicleShown = vehicle != nil && route != nil && vehicleStop != nil
+        ZStack(alignment: .top) {
+            VStack(spacing: 0) {
+                // Dummy vehicle card to space the stop list down exactly the height of the card
+                vehicleCardView(vehicle, vehicleStop, route)
```

Comment:
Yeah, we want the stops to scroll behind the vehicle card, so I used the `ZStack` to place the vehicle card above the `ScrollView`, but we then need the top of the stop list to be exactly at the bottom of the vehicle card, no matter how large it is, and since it's a ZStack, the layout of the top vehicle card doesn't change where the stop list is positioned, so I used a second one behind it to push it to the right height. But it looks like a ZStack isn't necessary and I can just use `.zIndex(1)` on the one in the VStack to get the same effect, good call!